### DISCUSSION TITLE
Google auth login flow redirects to login page when logging in with google, blocking access

### DISCUSSION THREAD

**Discussion by pjongeneel** [danny-avila/LibreChat#2596] [Troubleshooting] [answered]
### What happened?

When logging in with google, I would expect the redirect to send you to the landing page for new conversations. However, it seems to be redirecting to the login page again, and subsequent login attempts redirect in a loop. I'm able to create an account, as I can see that the account gets added to the user table, however, login attempts do not work.

Version: v0.7.1
Deployment: ubuntu docker compose 

Relevant .env context
```
HOST=0.0.0.0
PORT=3081
MONGO_URI=mongodb://127.0.0.1:27017/LibreChat
DOMAIN_CLIENT=http://xxx.compute.amazonaws.com:3081
DOMAIN_SERVER=http://xxx.compute.amazonaws.com:3081
ALLOW_SOCIAL_LOGIN=true
ALLOW_SOCIAL_REGISTRATION=true
GOOGLE_CLIENT_ID=xxxxx
GOOGLE_CLIENT_SECRET=xxxxxx
GOOGLE_CALLBACK_URL=/oauth/google/callback
```

![image](https://github.com/danny-avila/LibreChat/assets/28019025/43bff8a5-c2f5-4efc-a572-76a80ffd2c25)

### Steps to Reproduce

Follow guide on how to auth with google
Deploy librechat after .env configuration updates
Attempt to continue with google on deployment

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by danny-avila** [comment]
it works, something's up with your setup. Will check back in when I can help 

https://github.com/danny-avila/LibreChat/assets/110412045/6b16180a-1940-49fc-8730-bfe7afaf3541

**Reply by danny-avila** [reply]
at first glance, are you sure you need the port in the domain name (for client and server)?

DOMAIN_CLIENT=http://xxx.compute.amazonaws.com:3081

Seems a bit odd, try omitting it.

DOMAIN_CLIENT=http://xxx.compute.amazonaws.com

**Reply by pjongeneel** [reply]
Tried omitting, and getting the same result. 

If I run `sudo docker compose -f ./deploy-compose.yml up` and then try to access my server, I do get the login-page. I can also log in with a regular account. However, when refreshing the page from within the app, it immediately logs me out and sends me back to the login page. I think this is the root of the issue, because the google auth flow redirects to the login page, and I'm assuming it's doing the same thing as when I refresh using a standard account ("removing the login token and redirecting me to the login page"). 

If I create an ssh tunnel to my instance and use "localhost:3080" instead, everything works fine. The refresh issue also goes away. 

It's as simple as (with ssh tunnel) set DOMAIN_CLIENT=localhost:3080 - works
                                                      set DOMAIN_CLIENT=ipaddress:3080 (with and without port) - refresh issue (which is what I think causes the google auth flow to redirect back to the login page)

**Reply by danny-avila** [reply]
ok i think its lack of HTTPS in the protocol, what you are describing is expected when the connection is not secure. I'm going to test social login in an https environment soon to confirm but feeling this is it.

**Reply by pjongeneel** [reply]
awesome - thanks for the quick reply. yeah I have a feeling that might be it, will try to set up https on my end when I get the chance and see if that helps.

**Reply by pjongeneel** [reply, marked as answer]
can confirm - adding SSL works

**Comment by jroakes** [comment]
We are having a similar issue as this one, but in our case, we are using a CDN (Cloudfront).  Origin is a SSLd subdomain for a hosting provider, and the CDN handles the main domain (e.g. chat.domain.com) and SSL.  I assume there is some issue with Google getting the public URL and the origin URL coming though the backend on Librechat.  

I wanted to see if there was a setting I was missing for CDN URLs / etc to handle this.
